source/app/libpd.c

```diff
@@ -171,14 +180,40 @@ static void parodus_receive()
                         wrp_free_struct (res_wrp_msg);
                     }
             }
+
+            //handle cloud-status retrieve response received from parodus
+            if (wrp_msg->msg_type == WRP_MSG_TYPE__RETREIVE)
+            {
+				sourceService = wrp_get_msg_element(WRP_ID_ELEMENT__SERVICE, wrp_msg, SOURCE);
+				sourceApplication = wrp_get_msg_element(WRP_ID_ELEMENT__APPLICATION, wrp_msg, SOURCE);
+
+				if(sourceService != NULL && sourceApplication != NULL && strcmp(sourceService,"parodus")== 0 && strcmp(sourceApplication,"cloud-status")== 0)
+				{
+					WalInfo("cloud-status Retrieve response received from parodus : %s len %lu\n",(char *)wrp_msg->u.crud.payload, strlen(wrp_msg->u.crud.payload) );
+
+					parsePayloadForStatus(wrp_msg->u.crud.payload, &status);
+					if(status !=NULL)
+					{
+						//set this as global conn status. add lock before update it.
+						set_global_cloud_status(status);
+						WalInfo("set cloud-status value as %s\n", status);
+
+						pthread_mutex_lock (&cloud_mut);
+						WalPrint("mutex lock in producer thread\n");
+						pthread_cond_signal(&cloud_con);
```

Comment:
I feel it is better to move the pthread_cond_signal and pthread_cond_wait also into the getter and setter functions so it is better abstraction.